⚠️ Nightly upstream-dev CI failed ⚠️

[Workflow Run URL](https://github.com/pydata/xarray/actions/runs/12267044583)
<details><summary>Python 3.12 Test Summary</summary>

```
xarray/tests/test_cftimeindex.py::test_multiindex: KeyError: '2001-01'
xarray/tests/test_formatting.py::test_display_nbytes: AssertionError: assert '<xarray.Data...197 1198 1199' == '<xarray.Data...197 1198 1199'
  
  Skipping 86 identical leading characters in diff, use -v to show
  - 8, 1199], dtype=int16)
  + 8, 1199], shape=(1200,), dtype=int16)
  ?          +++++++++++++++
    Coordinates:
      * foo      (foo) int16 2kB 0 1 2 3 4 5 6 ... 1194 1195 1196 1197 1198 1199
```

</details>

> xarray/tests/test_formatting.py::test_display_nbytes: AssertionError: assert '<xarray.Data...197 1198 1199' == '<xarray.Data...197 1198 1199'

it looks like `numpy` changed their `repr` to sometimes include the `shape`, which I'm not sure we'd want? Not sure, though, because this only affects the data section of `Variable` / `DataArray`.

Other than that, we have a bunch of `zarr` errors:
```
xarray/tests/test_backends.py::TestZarrDictStore::test_manual_chunk[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrDictStore::test_warning_on_bad_chunks[2]: Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
xarray/tests/test_backends.py::TestZarrDictStore::test_write_uneven_dask_chunks[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrDictStore::test_encoding_chunksizes[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrDirectoryStore::test_manual_chunk[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrDirectoryStore::test_warning_on_bad_chunks[2]: Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
xarray/tests/test_backends.py::TestZarrDirectoryStore::test_write_uneven_dask_chunks[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrDirectoryStore::test_encoding_chunksizes[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrWriteEmpty::test_manual_chunk[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrWriteEmpty::test_warning_on_bad_chunks[2]: Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
xarray/tests/test_backends.py::TestZarrWriteEmpty::test_write_uneven_dask_chunks[2]: ValueError: ndarray is not C-contiguous
xarray/tests/test_backends.py::TestZarrWriteEmpty::test_encoding_chunksizes[2]: ValueError: ndarray is not C-contiguous
```

three `dask` / `distributed` errors:
```
xarray/tests/test_dask.py::TestToDaskDataFrame::test_to_dask_dataframe: AssertionError: DataFrame.index are different

Attribute "dtype" are different
[left]:  object
[right]: StringDtype(storage=pyarrow, na_value=<NA>)
xarray/tests/test_distributed.py::test_async: IndexError: tuple index out of range
xarray/tests/test_variable.py::TestVariableWithDask::test_multiindex: ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

a `CFTimeIndex` error:
```
xarray/tests/test_cftimeindex.py::test_multiindex: KeyError: '2001-01'
```

and two datetime + `dask` errors:
```
xarray/tests/test_variable.py::TestVariableWithDask::test_datetime64_conversion: assert True == False
xarray/tests/test_variable.py::TestVariableWithDask::test_timedelta64_conversion: assert True == False
```

--- a/xarray/tests/test_cftimeindex.py
+++ b/xarray/tests/test_cftimeindex.py
@@ -1188,6 +1188,7 @@ def test_to_datetimeindex_feb_29(calendar):
         index.to_datetimeindex()
 
 
+@pytest.mark.xfail(reason="fails on pandas main branch")
 @requires_cftime
 def test_multiindex():
     index = xr.cftime_range("2001-01-01", periods=100, calendar="360_day")

--- a/xarray/tests/test_formatting.py
+++ b/xarray/tests/test_formatting.py
@@ -1020,9 +1020,10 @@ Data variables:
     assert actual == expected
 
     actual = repr(xds["foo"])
-    expected = """
+    array_repr = repr(xds.foo.data).replace("\n     ", "")
+    expected = f"""
 <xarray.DataArray 'foo' (foo: 1200)> Size: 2kB
-array([   0,    1,    2, ..., 1197, 1198, 1199], dtype=int16)
+{array_repr}
 Coordinates:
   * foo      (foo) int16 2kB 0 1 2 3 4 5 6 ... 1194 1195 1196 1197 1198 1199
 """.strip()

